scripts/build/build/target.py

```diff
@@ -285,8 +285,64 @@ def HumanString(self):
 
         return result
 
+    def CompletionStrings(self, value: str) -> List[str]:
+        """Get a list of completion strings for this target."""
+
+        if self.name.startswith(value):
+            return [self.name + "-"]
+        if not value.startswith(self.name + "-"):
+            return []
+
+        completions = []
+        prefix = self.name
+        suffix = value[len(self.name) + 1:]
+
+        for i, targets in enumerate(self.fixed_targets):
+            for target in targets:
+                if suffix.startswith(target.name + "-"):
+                    # Strip the prefix and continue processing.
+                    prefix += f"-{target.name}"
+                    suffix = suffix[len(target.name) + 1:]
+                    break
+            else:
+                for target in targets:
+                    # NOTE: We are not validating whether the target is acceptable
+                    #       for the given value. Some validation rules require the
+                    #       full target name to be known, so in case of completions
+                    #       we just assume that the target is acceptable.
+                    completions.append(f"{prefix}-{target.name}")
+                # If we are not processing the last fixed target or we have
+                # some modifiers defined, append "-" to the generated completions.
+                if i < len(self.fixed_targets) - 1 or self.modifiers:
```

Comment:
I have to check whether modifiers should be taken into account here, because we can build example without any modifiers. It's a blocking comment just in case.